src/main/java/com/pusher/rest/util/Crypto.java

```diff
@@ -0,0 +1,105 @@
+package com.pusher.rest.util;
+
+import com.google.gson.Gson;
+import org.abstractj.kalium.crypto.Random;
+import org.abstractj.kalium.crypto.SecretBox;
+
+import java.nio.charset.StandardCharsets;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+
+/**
+ * Crypto is a class containing all the utility methods needed to do End to End Encryption for End to End Encrypted Channels.
+ * Read more here: https://pusher.com/docs/client_api_guide/client_encrypted_channels
+ */
+public class Crypto {
+    /**
+     * The prefix that is used to identify End to End Encrypted Channels
+     */
+    private static final String PRIVATE_ENCRYPTED_CHANNEL_PREFIX = "private-encrypted-";
+
+    /**
+     * The GSON instance to be used for serialisation
+     */
+    private final Gson BODY_SERIALISER;
+
+    /**
+     * 32 character long secret you define for end to end encryption
+     */
+    private final String ENCRYPTION_MASTER_KEY;
+
+    /** Construct an instance of the Crypto object which will allow for encryption of Pusher Channels payloads.
+     * @param encryptionMasterKey The Encryption Master Key, a 32 character long secret you define for end to end encryption
+     * @param serialiser The GSON instance for serialisation of the payload
+     * @throws RuntimeException if Libsodium isn't available on the platform
+     */
+    public Crypto(String encryptionMasterKey, Gson serialiser) throws RuntimeException {
+        this.BODY_SERIALISER = serialiser;
+        this.ENCRYPTION_MASTER_KEY = encryptionMasterKey;
+        if(!cryptoAvailable()) {
+            throw new RuntimeException("The Pusher client requires Libsodium for End to End Encryption");
```

Comment:
I _think_ this should throw the same exception that kalium does, i.e. `UnsatisfiedLinkError`. WDYT?

Reply:
I personally prefer RuntimeException since its consistent with other exceptions we throw, but don't feel massively strongly so happy to change if you want me to

Reply:
Sure. If it's consistent with other similar exceptions we throw, and then I'd rather stick to what it is for consistency.